peers.go

```diff
@@ -47,23 +47,37 @@ type NoPeers struct{}
 func (NoPeers) PickPeer(key string) (peer ProtoGetter, ok bool) { return }
 
 var (
-	portPicker func() PeerPicker
+	portPicker func(groupName string) PeerPicker
 )
 
 // RegisterPeerPicker registers the peer initialization function.
 // It is called once, when the first group is created.
+// Either RegisterPeerPicker or RegisterPerGroupPeerPicker should be
+// called exactly once, but not both.
 func RegisterPeerPicker(fn func() PeerPicker) {
 	if portPicker != nil {
 		panic("RegisterPeerPicker called more than once")
 	}
+	portPicker = func(_ string) PeerPicker { return fn() }
+}
+
+// RegisterPerGroupPeerPicker registers the peer initialization function,
+// which takes the groupName, to be used in choosing a PeerPicker.
+// It is called once, when the first group is created.
+// Either RegisterPeerPicker or RegisterPerGroupPeerPicker should be
+// called exactly once, but not both.
+func RegisterPerGroupPeerPicker(fn func(groupName string) PeerPicker) {
+	if portPicker != nil {
+		panic("RegisterPeerPicker called more than once")
```

Comment:
Misleading panic text. It implies that just `RegisterPeerPicker` called more that once, while in reality it was `RegisterPerGroupPeerPicker` called twtice or both `RegisterPeerPicker` & `RegisterPerGroupPeerPicker` called. It can be even worse inside the `RegisterPeerPicker()` - people will try to search for duplicate `RegisterPeerPicker()` calls while it was  `RegisterPeerPicker() + RegisterPerGroupPeerPicker()`.